sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_http_logging_policy.py

```diff
@@ -41,213 +43,303 @@
+                if multi_record:
+                    logger.info(client_settings, extra=cosmos_logger_attributes)
+                    logger.info(db_settings, extra=cosmos_logger_attributes)
+                    logger.info("Request URL: %r", redacted_url, extra=cosmos_logger_attributes)
+                    logger.info("Request method: %r", http_request.method, extra=cosmos_logger_attributes)
+                    logger.info("Request headers:", extra=cosmos_logger_attributes)
+                    for header, value in http_request.headers.items():
+                        value = self._redact_header(header, value)
+                        if value and value != HttpLoggingPolicy.REDACTED_PLACEHOLDER:
+                            logger.info("    %r: %r", header, value, extra=cosmos_logger_attributes)
+                    if isinstance(http_request.body, types.GeneratorType):
+                        logger.info("File upload", extra=cosmos_logger_attributes)
+                        return
+                    try:
+                        if isinstance(http_request.body, types.AsyncGeneratorType):
+                            logger.info("File upload", extra=cosmos_logger_attributes)
+                            return
+                    except AttributeError:
+                        pass
+                    if http_request.body:
+                        logger.info("A body is sent with the request", extra=cosmos_logger_attributes)
+                        return
+                    logger.info("No body was attached to the request", extra=cosmos_logger_attributes)
+                    return
+                log_string = client_settings
+                log_string += db_settings
+                log_string += "\nRequest URL: '{}'".format(redacted_url)
+                log_string += "\nRequest method: '{}'".format(http_request.method)
+                log_string += "\nRequest headers:"
+                for header, value in http_request.headers.items():
+                    value = self._redact_header(header, value)
+                    if value and value != HttpLoggingPolicy.REDACTED_PLACEHOLDER:
+                        log_string += "\n    '{}': '{}'".format(header, value)
+                if isinstance(http_request.body, types.GeneratorType):
+                    log_string += "\nFile upload"
+                    logger.info(log_string, extra=cosmos_logger_attributes)
+                    return
+                try:
+                    if isinstance(http_request.body, types.AsyncGeneratorType):
+                        log_string += "\nFile upload"
+                        logger.info(log_string, extra=cosmos_logger_attributes)
+                        return
+                except AttributeError:
+                    pass
+                if http_request.body:
+                    log_string += "\nA body is sent with the request"
+                    logger.info(log_string, extra=cosmos_logger_attributes)
```

Comment:
Wouldn't it be more efficient to collapse all these log lines into 1?

Reply:
Yes it would be, and we do that, but there is an environmental variable that can be set to make it into multiple logs that the parent class from core has. In order to keep the same functionality while allowing us to add filterable attributes I had to make sure i pass the same attribute values to all the logs that we make otherwise we would not return all the needed information.